ci-operator/step-registry/cucushift/upgrade/checktarget/cucushift-upgrade-checktarget-ref.metadata.json

```
@@ -0,0 +1,23 @@
+{
+	"path": "cucushift/upgrade/checktarget/cucushift-upgrade-checktarget-ref.yaml",
```

```suggestion
	"path": "cucushift/upgrade/check-not-recommended-upgrade/check-not-recommended-upgrade-ref.yaml",
```

How about this new name? @jiajliu @shellyyang1989 if you have more good idea, pls comment here.

---

check-conditional-update? and both of above also sgtm. 

I'm a bit conflicted now, shall we drop condition update? I don't remember why `--allow-not-recommended` var came out? At least, there should be users need it and then it's into our roadmap, right? As it is a supported var and documented in the official doc, then user might use it pending on their assessment. Of course, we could cover it in a dedicated ota case, but condition update is rare compared with regular upgrade, only regular ci could help cover most of condition updates in more paths.

---

@jiajliu I used to have the same concern, but as @shellyyang1989 said there is no user choose a not recommended version.
I read below doc again, it just mentioned `administrator can waive the safety guards and proceed the update`:  
https://docs.openshift.com/container-platform/4.15/updating/updating_a_cluster/updating-cluster-cli.html#update-conditional-upgrade-pathupdating-cluster-cli

---

so the doc means there will be users update with --allow-not-recommended, right?

---

I meant updating to a not recommended image is supported but I don't think it's a common option that's worth to be covered in common upgrade step. We do have test case to cover the scenario. We can cover it in test case level.

---

Back to the topic of the step naming, because `checktarget` is a bit common word, not easy to understanding what purpose for this step from its name, so that is why I suggest to rename it to `check-not-recommended`.